### PR TITLE
rx_didChangeState Throws #4 issue fixes

### PR DESCRIPTION
**Pull request by sergdort**

### CODE REVIEW

**Comment by sergdort** [comment]
Added test to cover this problem

File: Pod/Classes/MKMapView+Rx.swift [hunk 1]
```diff
@@ -176,8 +176,13 @@ extension MKMapView {
...
                 return (view: try castOrThrow(MKAnnotationView.self, a[1]),
-                    newState: try castOrThrow(MKAnnotationViewDragState.self, a[2]),
-                    oldState: try castOrThrow(MKAnnotationViewDragState.self, a[3]))
+                    newState: try castOrThrow(UInt.self, a[2]),
+                    oldState: try castOrThrow(UInt.self, a[3]))
+            }
+            .map { (view, newState, oldState) in
+                return (view: view,
+                    newState: MKAnnotationViewDragState(rawValue: newState)!,
+                    oldState: MKAnnotationViewDragState(rawValue: oldState)!)
```
Comment (sergdort) [inline Pod/Classes/MKMapView+Rx.swift:185]: The second `map` operator looks to not so ugly, if I would just cast it the first one

What do you think?

**Comment by sger** [comment]
nice @sergdort

**Comment by sergdort** [comment]
And what do you think about to add some unit tests to cove the rest of the logic?
Just like in `RxCocoa` test

I can spend some time to contribute

**Comment by sger** [comment]
that sounds good i'm going to create a new cocoapod version also.

**Pull request merged**